src/hazelcore/dynamics/EvalState.re

```diff
@@ -0,0 +1,16 @@
+[@deriving sexp]
+type t = (EvalEnvId.t, EvalStats.t);
```

Comment:
Make this a record so we can use update syntax.

Reply:
Makes sense. I think I used a tuple because there doesn't seem to be much use of records in the project (or at least in hazelcore), but that's not a very good reason to avoid it.

`Result.t` could also benefit from record syntax